Tests/Opc.Ua.Security.Certificates.Tests/CRLTests.cs

```diff
@@ -0,0 +1,188 @@
+        [Test]
+        public void CrlInternalBuilderTest()
+        {
+            var dname = new X500DistinguishedName("CN=Test");
+            var hash = HashAlgorithmName.SHA256;
+            var crlBuilder = new CrlBuilder(dname, hash)
+                .SetNextUpdate(DateTime.Today.AddDays(30));
+            byte[] serial = new byte[] { 4, 5, 6, 7 };
+            var revokedarray = new RevokedCertificate(serial);
+            crlBuilder.RevokedCertificates.Add(revokedarray);
+            string serstring = "45678910";
+            var revokedstring = new RevokedCertificate(serstring);
+            crlBuilder.RevokedCertificates.Add(revokedstring);
+            crlBuilder.CrlExtensions.Add(X509Extensions.BuildCRLNumber(123));
+            var crlEncoded = crlBuilder.Encode();
+            Assert.NotNull(crlEncoded);
+            var x509Crl = new X509CRL();
+            x509Crl.DecodeCrl(crlEncoded);
+            Assert.NotNull(x509Crl);
+            Assert.NotNull(x509Crl.CrlExtensions);
+            Assert.NotNull(x509Crl.RevokedCertificates);
+            Assert.AreEqual(dname.RawData, x509Crl.IssuerName.RawData);
+            //Assert.AreEqual(crlBuilder.ThisUpdate, x509Crl.ThisUpdate);
+            //Assert.AreEqual(crlBuilder.NextUpdate, x509Crl.NextUpdate);
+            Assert.AreEqual(2, x509Crl.RevokedCertificates.Count);
+            Assert.AreEqual(serial, x509Crl.RevokedCertificates[0].UserCertificate);
+            Assert.AreEqual(serstring, x509Crl.RevokedCertificates[1].SerialNumber);
+            Assert.AreEqual(1, x509Crl.CrlExtensions.Count);
+            Assert.AreEqual(hash, x509Crl.HashAlgorithmName);
+        }
+
+        /// <summary>
+        /// Validate the internal CRL encoder and decoder pass.
+        /// </summary>
+        [Test]
+        public void CrlBuilderTest()
+        {
+            var dname = new X500DistinguishedName("CN=Test");
+            var hash = HashAlgorithmName.SHA256;
+            var crlBuilder = new CrlBuilder(dname, hash)
+                .SetNextUpdate(DateTime.Today.AddDays(30));
+            byte[] serial = new byte[] { 4, 5, 6, 7 };
+            var revokedarray = new RevokedCertificate(serial);
+            crlBuilder.RevokedCertificates.Add(revokedarray);
+            string serstring = "45678910";
+            var revokedstring = new RevokedCertificate(serstring);
+            crlBuilder.RevokedCertificates.Add(revokedstring);
+            crlBuilder.CrlExtensions.Add(X509Extensions.BuildCRLNumber(123));
+            var crlEncoded = crlBuilder.Encode();
+            Assert.NotNull(crlEncoded);
+            var x509Crl = new X509CRL();
+            x509Crl.DecodeCrl(crlEncoded);
+            Assert.NotNull(x509Crl);
+            Assert.NotNull(x509Crl.CrlExtensions);
+            Assert.NotNull(x509Crl.RevokedCertificates);
+            Assert.AreEqual(dname.RawData, x509Crl.IssuerName.RawData);
+            //Assert.AreEqual(crlBuilder.ThisUpdate, x509Crl.ThisUpdate);
```

Comment:
Why is this code is commented out? If not valid check, than please remove it completely

Reply:
need to check why its not passing...

Reply:
fixed (i think)